app/tests/cases_tests/test_api.py

```diff
@@ -295,21 +295,55 @@ def test_image_file_post_permissions(client, is_active, expected_response):
 
 
 @pytest.mark.django_db
-def test_process_images_api_view(client):
+def test_process_images_api_view(client, settings):
+    # Override the celery settings
+    settings.task_eager_propagates = (True,)
+    settings.task_always_eager = (True,)
+
     user = UserFactory()
-    ai = AlgorithmImageFactory(creator=user)
-    ai.algorithm.add_user(user)
-    upload_session = RawImageUploadSessionFactory(
-        creator=user, algorithm_image=ai
-    )
+
+    us = RawImageUploadSession(creator=user)
+    us.save()
+    ri = RawImageFile(upload_session=us)
+    ri.save()
 
     response = get_view_for_user(
         viewname="api:upload-session-process-images",
-        reverse_kwargs={"pk": upload_session.pk},
+        reverse_kwargs={"pk": us.pk},
         user=user,
         client=client,
         method=client.patch,
         content_type="application/json",
     )
     assert response.status_code == 200
-    assert "Images are uploaded." in str(response.content)
+
+    ri.refresh_from_db()
+    ri.consumed = True
+    ri.save()
+
+    response = get_view_for_user(
+        viewname="api:upload-session-process-images",
+        reverse_kwargs={"pk": us.pk},
+        user=user,
+        client=client,
+        method=client.patch,
+        content_type="application/json",
+    )
+    assert response.status_code == 400
+
+    ri.refresh_from_db()
+    ri.consumed = False
+    ri.save()
+    us.refresh_from_db()
+    us.session_state = "created"
+    us.save()
+
+    response = get_view_for_user(
+        viewname="api:upload-session-process-images",
+        reverse_kwargs={"pk": us.pk},
+        user=user,
+        client=client,
+        method=client.patch,
+        content_type="application/json",
+    )
+    assert response.status_code == 400
```

Comment:
This test is just testing if the post save task on `us.save()` works, which we know it does as it is tested elsewhere. What are you intending to test here? Maybe it was restarting the job, in which case 

```
    us.refresh_from_db()
    us.session_state = "created"
    us.save()
```

needs to happen before

```
    ri.refresh_from_db()
    ri.consumed = False
    ri.save()
```

and then 

```
    assert response.status_code == 200
```

as the job will be restarted in this case.

Reply:
@jmsmkn 
This gives me a 400 response. Why should it be 200? For that, us.session_state must be "stopped". Isn't it?

Reply:
I don't understand what you're trying to do here then, what is the intention of this test?

Reply:
I guess the last test is superfluous. I will remove it.